File: 2022/tests/day12.test.ts
import { readFileSync } from 'fs';
import { part1, part2 } from '../src/day12';

const dayNumber = __filename.match(/day(\d+)\.test\.ts/)?.[1];
const inputFile: string = `input/day${dayNumber}.txt`;

const example: string = `
Sabqponm
abcryxxl
accszExk
acctuvwj
abdefghi
`.trim();
const input: string = readFileSync(inputFile, 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example)).toBe(31);
  });
  test(`solution`, () => {
    expect(part1(input)).toBe(534);
  });
});

describe('part 2', () => {
  test('example', () => {
    expect(part2(example)).toBe(29);
  });
  test(`solution`, () => {
    expect(part2(input)).toBe(525);
  });
});

File: 2025/test/advent_of_code/day_12_test.exs
defmodule AdventOfCode.Day12Test do
  use ExUnit.Case

  import AdventOfCode.Day12

  @day 12
  @year 2025

  @example String.trim("""
           0:
           ###
           ##.
           ##.

           1:
           ###
           ##.
           .##

           2:
           .##
           ###
           ##.

           3:
           ##.
           ###
           ##.

           4:
           ###
           #..
           ###

           5:
           ###
           .#.
           ###

           4x4: 0 0 0 0 2 0
           12x5: 1 0 1 0 2 2
           12x5: 1 0 1 0 3 2
           """)

  test "part1 example" do
    result = part1(@example)

    assert result == 1 # 2
  end

  test "part1" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part1(input)

    assert result == 460
  end
end
